### ISSUE TITLE
Documentation's NumberRangeFilter doesn't support ranges with one bound

### ISSUE THREAD

**Issue by dunk** [carltongibson/django-filter#1048]
The [NumberRangeFilter](https://django-filter.readthedocs.io/en/master/ref/filters.html#baserangefilter) suggested in the `BaseRangeFilter` documentation doesn't support "greater than" and "less than" style ranges (i.e. those with just a lower or upper bound, but not both), because `BaseRangeFilter` [depends on BaseRangeField](https://github.com/carltongibson/django-filter/blob/82a47fb7bbddedf179f110723003f3b28682d7fe/django_filters/filters.py#L545), and the `BaseRangeField` implementation [forces a length of 2](https://github.com/carltongibson/django-filter/blob/82a47fb7bbddedf179f110723003f3b28682d7fe/django_filters/fields.py#L196).

This is a bit of an inconsistent API choice because the [NumericRangeFilter](https://django-filter.readthedocs.io/en/master/ref/filters.html#numericrangefilter), which is also a type of "range", _does_ [does support this kind of range](https://github.com/carltongibson/django-filter/blob/82a47fb7bbddedf179f110723003f3b28682d7fe/django_filters/filters.py#L368). While I appreciate that these are quite different under the hood (relying on Postgres range types), that shouldn't matter to the user of django-filter's API, who will in both cases often want to do all three of "greater than", "less than" and "between".

**Comment by carltongibson** [comment]
Yep, unbounded range should be allowed. Fancy adding a test and fix?

**Comment by rpkilby** [comment]
I'm kind of -0 on adding unbounded ranges. This does make sense for `NumericRangeFilter`, which targets more traditional forms usage, but `BaseRangeFilter` is CSV-based and more suitable for APIs. Instead of adding complexity to the `BaseRangeFilter`, I'd argue that users should use separate `gt`/`gte`/`lt`/`lte` filters.

That said, above opinion is -0 / gut reaction.
